roc/address.go

```diff
@@ -0,0 +1,73 @@
+package roc
+
+/*
+#cgo LDFLAGS: -lroc
+#include <roc/receiver.h>
+#include <roc/address.h>
+#include <roc/sender.h>
+#include <roc/log.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"runtime"
+	"unsafe"
+)
+
+func stringToCharArray(str string) []C.char {
+	charArray := make([]C.char, len(str))
+	for ind, r := range str {
+		charArray[ind] = (C.char)(r)
+	}
+	return charArray
+}
+
+// NewAddress parses the `ip`, `port` and `family` and initializes the Address object
+func NewAddress(family Family, ip string, port int) (*Address, error) {
+	a := new(Address)
+	a.mem = make([]byte, C.sizeof_roc_address)
+	a.raw = (*C.roc_address)(unsafe.Pointer(&a.mem))
+
+	cfamily := (C.roc_family)(family)
+	ip = safeString(ip)
+	cip := stringToCharArray(ip)
+	cport := (C.int)(port)
+	errCode := C.roc_address_init(a.raw, cfamily, (*C.char)(unsafe.Pointer(&cip[0])), cport)
+	runtime.KeepAlive(a.mem)
```

Comment:
Hm, after more thought and googling, it seems that we don't need `runtime.KeepAlive` here.

Why:

* `unsafe.Pointer` references are traceable by GC
* so `*C.roc_address` should be traceable as well
* cgo automatically inserts `cgoUse()` call for every pointer argument, which is superior to `runtime.KeepAlive` (in addition, it forces the pointer to escape to heap, to prevent it moving during stack reallocation)

Thus, `a.mem` is guaranteed to be alive during `C.roc_address_init` call.

Why is this important? Because if it was wrong, we had to add `runtime.KeepAlive` to all other cgo calls, for every argument passed to C. And it seems that we don't have to, in most cases.

Reply:
cool!